scylla/src/transport/load_balancing/token_aware.rs

```diff
@@ -138,7 +140,20 @@ impl LoadBalancingPolicy for TokenAwarePolicy {
                     "TokenAware"
                 );
 
-                self.child_policy.apply_child_policy(replicas)
+                let fallback_plan = {
+                    let replicas_set: HashSet<SocketAddr> =
+                        replicas.iter().map(|node| node.address).collect();
+
+                    self.child_policy
+                        .plan(&Statement::empty(), cluster)
+                        .filter(move |node| !replicas_set.contains(&node.address))
```

Comment:
we could even drop the filter and ignore the fact that we get duplicates, but this one is quite elegant, so let's keep it